tests/test_gencast_with_thermalizer.py

```diff
@@ -0,0 +1,187 @@
+import torch
+
+from graph_weather.models.forecast import GraphWeatherForecaster
+
+
+def test_gencast_thermal_integration():
+    # Test with a larger grid to properly test UNet capabilities
+    # Create a 3x3 grid (9 nodes)
+    lat_lons = [(i // 3, i % 3) for i in range(9)]  # 3x3 grid
+
+    print(f"Number of lat_lons: {len(lat_lons)}")
+    print("Grid should be: 3x3")
+
+    model = GraphWeatherForecaster(
+        lat_lons,
+        use_thermalizer=True,
+        feature_dim=3,
+        aux_dim=0,
+        node_dim=256,  # Match the thermalizer input_dim
+        num_blocks=1,  # Reduce complexity for testing
+    )
+
+    print(f"Model grid_shape: {model.grid_shape}")
+
+    features = torch.randn(1, len(lat_lons), 3)
+
+    print(f"Input features shape: {features.shape}")
+
+    try:
+        # Run GenCast to get one-step prediction
+        pred = model(features, t=torch.randint(0, 1000, (1,)).item())
+
+        # Check output shape - should match input
+        print(f"Input shape: {features.shape}")
+        print(f"Output shape: {pred.shape}")
+
+        # The output might be in grid format if constraints are applied
+        if pred.dim() == 4:  # [B, C, H, W] format
+            assert pred.shape[0] == features.shape[0]  # Same batch size
+            assert (
+                pred.shape[2] * pred.shape[3] == features.shape[1]
+            )  # Same number of spatial points
+            print("Output is in grid format [B, C, H, W]")
+        else:  # [B, N, C] format
+            assert pred.shape == features.shape
+            print("Output is in node format [B, N, C]")
+
+        # Check for valid outputs
+        assert not torch.isnan(pred).any(), "Output contains NaN values"
+        assert torch.isfinite(pred).all(), "Output contains infinite values"
+
+        print("Test passed!")
+
+    except Exception as e:
```

Comment:
There shouldn't be a try/except in tests, if this causes an error, that's the point of the test to catch, and if this should cause an error, we can test for that, but otherwise, the test should either pass or not, not do a try/except.